src/web/src/components/SearchInput/SearchInput.tsx

```diff
@@ -5,7 +5,7 @@ import useSearchValue from '../../hooks/use-search-value';
 const SearchInput = () => {
   const { filter } = useSearchValue();
 
-  return filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />;
+  return <>{filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />}</>;
```

Comment:
Can you help me understand more about this change?  What does wrapping in an element do for us in this case that returning the component won't?

Reply:
@humphd, `filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />;` is like an function that's called in every re-render, and its return result in unknown. When either `AuthorSearchInput` or `PostSearchInput` is returned, the current input is unmounted and re-rendered so the input loses focus without `autofoucs` attribute.


`<>{filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />}</>` is a React Component that's has conditional rendering, `AuthorSearchInput` and `PostSearchInput` won't re-mount if it doesn't change.

I believe that's how React works, let me tag Dan Abramov to confirm :).

Reply:
In both cases the React component tree will be different, whether at the root or child level. "filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />; is like an function" I don't think so.  It's the same expression you're evaluating in `<>{filter === 'author' ? <AuthorSearchInput /> : <PostSearchInput />}</>`

```js
function ComponentA() { return <div></div>; }
function ComponentB() { return <div></div>; }

const x = 1;

const element = <>{x === 1 ? <ComponentA /> : <ComponentB />}</>;
// var element = /*#__PURE__*/React.createElement(React.Fragment, null, x === 1 ? /*#__PURE__*/React.createElement(ComponentA, null) : /*#__PURE__*/React.createElement(ComponentB, null));

const element2 = x === 1 ? <ComponentA /> : <ComponentB />;
// var element2 = x === 1 ? /*#__PURE__*/React.createElement(ComponentA, null) : /*#__PURE__*/React.createElement(ComponentB, null);
```

If there's a real difference when wrapping in a fragment, related to re-rendering, I'd like to understand what it is. Both are going to evaluate to, and produce the same DOM.

Reply:
Are you saying that the fragment provides stability between re-renders in order to maintain focus?

Reply:
> Are you saying that the fragment provides stability between re-renders in order to maintain focus?

Yes, I mean that.
The difference is that the React fragment prevents unmounting, both will produce the same DOM, but the first one doesn't re-mount and the second one does on every re-render.

Reply:
OK, that makes sense.